geomesa-stream/geomesa-stream-datastore/src/main/scala/org/locationtech/geomesa/stream/datastore/StreamDataStore.scala

```diff
@@ -46,27 +54,27 @@ case class FeatureHolder(sf: SimpleFeature, env: Envelope) {
   }
 }
 
-class StreamDataStore(source: SimpleFeatureStreamSource, timeout: Int) extends ContentDataStore {
+class StreamDataStore(source: SimpleFeatureStreamSource, timeout: Int, ns: Option[String]) extends ContentDataStore {
   
   val sft = source.sft
   source.init()
   val qt = new SynchronizedQuadtree[SimpleFeature]
 
-  val cb =
-    Caffeine
-      .newBuilder()
-      .expireAfterWrite(timeout, TimeUnit.SECONDS)
-      .removalListener(
-        new RemovalListener[String, FeatureHolder] {
-          override def onRemoval(k: String, v: FeatureHolder, removalCause: RemovalCause): Unit = {
-            qt.remove(v.env, v.sf)
-          }
+  val cb = {
+    val builder = Caffeine.newBuilder()
+    if(timeout > 0) builder.expireAfterWrite(timeout, TimeUnit.SECONDS)
```

Comment:
[picky] space after if and i like curlies {}

Reply:
fixed